x/swaprouter/client/query_proto_wrap.go

```diff
@@ -106,3 +106,10 @@ func (q Querier) EstimateSwapExactAmountOut(ctx sdk.Context, req queryproto.Esti
 		TokenInAmount: tokenInAmount,
 	}, nil
 }
+
+// NumPools returns total number of pools.
+func (q Querier) NumPools(ctx sdk.Context, _ queryproto.NumPoolsRequest) (*queryproto.NumPoolsResponse, error) {
+	return &queryproto.NumPoolsResponse{
+		NumPools: q.K.GetNextPoolId(ctx) - 1,
```

Comment:
Consider changing the name from `NumPools` to `NextPoolId` since we potentially have plans to remove pools and then this query name could be misleading

Reply:
Seems that we are keeping pools. Also, I don't think we need to have a query for pool ids. I don't think any client would be interested in that. What clients might want to know is how many pools there are which we can infer from the next pool id

Reply:
Also, this is a direct port from gamm. I'm not being creative here. Let me know if you feel strongly though, we can discuss further